src/Microsoft.IdentityModel.Clients.ActiveDirectory/DeviceCodeResult.cs

```diff
@@ -77,5 +77,10 @@ internal DeviceCodeResult()
         /// Identifier of the target resource that would be the recipient of the token.
         /// </summary>
         public string Resource { get; internal set; }
+
+        /// <summary>
+        /// Identifier of the correlation id. Used in diagnostic purposes.
+        /// </summary>
+        public Guid CorrelationId { get; internal set; }
```

Comment:
>public Guid CorrelationId { get; internal set; } [](start = 8, length = 48)

I'm not sure we should expose this but not sure how otherwise to keep the code clean... perhaps this is fine. Jean-Marc?

Reply:
what is the concern, @henrik-me ? in term of code not being clean?

Reply:
I replied to quickly. No, @henrik-me : additive is fine provided we respect binary compatibility as well (in which case we are adding an extra compiler generated field. provided it's last that's fine)

Reply:
Q was also ensure you are fine adding CorrelationId here. Anyway I do not have any real concern.

---
In reply to: [175547217](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1017#discussion_r175547217) [](ancestors = 175547217)